tests/e2e/core-tests/specs/merchant/wp-admin-settings-shipping-zones.test.js

```diff
@@ -0,0 +1,119 @@
+		beforeAll(async () => {
+			await merchant.login();
+			await createSimpleProduct();
+		});
+
+		it('add shipping zone for San Francisco with free Local pickup', async () => {
+			// Add a new shipping zone for San Francisco 94107, CA, US with Local pickup
+			await addShippingZoneAndMethod(shippingZoneNameSF, california, sanFranciscoZIP, 'local_pickup');
+		});
+
+		it('add shipping zone for California with Free shipping', async () => {
+			// Add a new shipping zone for CA, US with Free shipping
+			await addShippingZoneAndMethod(shippingZoneNameFL, california, ' ', 'free_shipping');
+		});
+
+		it('add shipping zone for the US with Flat rate', async () => {
+			// Add a new shipping zone for the US with Flat rate
+			await addShippingZoneAndMethod(shippingZoneNameUS);
+
+			// Set Flat rate cost
+			await expect(page).toClick('a.wc-shipping-zone-method-settings', {text: 'Edit'});
+			await clearAndFillInput('#woocommerce_flat_rate_cost', '10');
+			await expect(page).toClick('button#btn-ok');
+			await merchant.logout();
+		});
+
+		it('allows customer to pay for a Flat rate shipping method', async() => {
+			await shopper.login();
+
+			// Add product to cart as a shopper
+			await shopper.goToShop();
+			await shopper.addToCartFromShopPage(simpleProductName);
+			await shopper.goToCart();
+
+			// Set shipping country to United States (US)
+			await expect(page).toClick('a.shipping-calculator-button');
+			await expect(page).toClick('#select2-calc_shipping_country-container');
+			await selectOptionInSelect2('United States (US)');
+
+			// Set shipping state to New York
+			await expect(page).toClick('#select2-calc_shipping_state-container');
+			await selectOptionInSelect2('New York');
+			await expect(page).toClick('button[name="calc_shipping"]');
+
+			// Verify shipping costs
+			await page.waitForSelector('.order-total');
+			await expect(page).toMatchElement('.shipping .amount', {text: '$10.00'});
+			await expect(page).toMatchElement('.order-total .amount', {text: '$19.99'});
+
+			await shopper.removeFromCart(simpleProductName);
+		})
+
+		it('allows customer to benefit from a Free shipping if in CA', async () => {
+			await shopper.goToShop();
+			await shopper.addToCartFromShopPage(simpleProductName);
+			await shopper.goToCart();
+
+			// Set shipping state to California
+			await expect(page).toClick('a.shipping-calculator-button');
+			await expect(page).toClick('#select2-calc_shipping_state-container');
+			await selectOptionInSelect2('California');
+
+			// Set shipping postcode to 94000
+			await clearAndFillInput('#calc_shipping_postcode', '94000');
+			await expect(page).toClick('button[name="calc_shipping"]');
+
+			// Verify shipping method and cost
+			await page.waitForSelector('.order-total');
+			await expect(page).toMatchElement('.shipping ul#shipping_method > li', {text: 'Free shipping'});
+			await expect(page).toMatchElement('.order-total .amount', {text: '$9.99'});
```

Comment:
What are your thoughts on pulling out the product pricing from the config, something like this?

```
const simpleProductPrice = config.has('products.simple.price') ? config.get('products.simple.price') : '9.99';
```

And then doing calculations with that price within the tests?

Reply:
I understand what you mean, but the reason why I do with manually entering values is to keep less amount of code, and because this was just a simple checking for prices. I did already that way in some other test and yes it was just fine to have calculation + value from config, but in this case, I would rather keep it without.

Reply:
That makes sense, my concern is that we are pulling in the price from the config in `components.js`, e.g.:

```
const simpleProductPrice = config.has('products.simple.price') ? config.get('products.simple.price') : '9.99';
```

Which is then used in the `createSimpleProduct()` call, as below:

```
const createSimpleProduct = async () => {
	const product = await factories.products.simple.create( {
		name: simpleProductName,
		regularPrice: simpleProductPrice
	} );
	return product.id;
} ;
```

So if that were to change, that would affect the test results.

Reply:
You were right. I just updated my test.